index.bs

```diff
@@ -74,6 +75,125 @@ A user agent must not <a spec=HTML>delay the load event</a> of the <{link}> elem
+    1. If |rule| is null, then [=iteration/continue=].
+    1. [=list/Append=] |rule| to |result|'s [=speculation rule set/prefetch rules=].
+  1. Return |result|.
+</div>
+
+<div algorithm="parse a speculation rule">
+  To <dfn export>parse a speculation rule</dfn> given a [=map=] |input| and a [=URL=] |baseURL|, perform the following steps. They return a [=speculation rule=] or null.
+
+  1. If |input| has any [=map/key=] other than "`source`", "`urls`", and "`requires`", then return null.
+  1. If |input|["`source`"] does not [=map/exist=] or is not the [=string=] "`list`", then return null.
+  1. Let |urls| be an empty [=list=].
+  1. If |input|["`urls`"] does not [=map/exist=], is not a [=list=], or has any element which is not a [=string=], then return null.
+  1. [=list/For each=] |urlString| of |input|["`urls`"]:
+    1. Let |parsedURL| be the result of [=basic URL parser|parsing=] |urlString| with |baseURL|.
+    1. If |parsedURL| is failure, then [=iteration/continue=].
+    1. If |parsedURL|'s [=url/scheme=] is not an [=HTTP(S) scheme=], then [=iteration/continue=].
+    1. [=list/Append=] |parsedURL| to |urls|.
+  1. Let |requirements| be an empty [=ordered set=].
+  1. If |input|["`requires`"] [=map/exists=], but is not a [=list=], then return null.
+  1. [=list/For each=] |requirement| of |input|["`requires`"]:
+    1. If |requirement| is not the [=string=] "`anonymous-client-ip`", then return null.
+    1. [=set/Append=] |requirement| to |requirements|.
+  1. Return a [=speculation rule=] with [=speculation rule/URLs=] |urls| and [=speculation rule/requirements=] |requirements|.
+</div>
+
+<h3 id="speculation-rules-processing">Processing Model</h3>
+
+Periodically, the user agent may [=queue a task=] to the [=DOM manipulation task source=] to [=consider speculation=] for any [=document=].
+
+<div class="note">
+  The user agent will likely do this after the insertion of new speculation rules, or when resources are idle and available.
+</div>
+
+<div algorithm="consider speculation">
+  To <dfn export>consider speculation</dfn> for a [=document=] |document|:
+
+  1. For each <{script}> element [=connected=] to |document| where <a spec=html>the script's type</a> is "`speculationrules`" and [=the script's result=] is a [=speculation rule set=]:
+    1. [=list/For each=] |rule| of [=the script's result=]:
+      1. Let |requiresAnonymousClientIP| be true if |rule|'s [=speculation rule/requirements=] [=set/contains=] "`anonymous-client-ip`", and false otherwise.
+      1. [=list/For each=] |url| of |rule|'s [=speculation rule/URLs=]
+        1. The user agent MAY prefetch |url| given |requiresAnonymousClientIP|.
+           <div class="issue">TODO: expand this to actually elaborate on how prefetch works, once initiated, and to incorporate the |requiresAnonymousClientIP| flag.</div>
+</div>
+
+<div class="issue">
```

Comment:
I think it's more common to have `<p class="issue">`, or at least it is more consistent with the notes in this spec. Also feel free to omit the end tag like other notes/issues here.

Reply:
done